quic/s2n-quic-core/src/event/generated.rs

```diff
@@ -1161,9 +1199,16 @@ pub mod api {
             builder::Frame::Ping {}
         }
     }
-    impl<AckRanges> IntoEvent<builder::Frame> for &crate::frame::Ack<AckRanges> {
+    impl<AckRanges: crate::frame::ack::AckRanges> IntoEvent<builder::Frame>
+        for &crate::frame::Ack<AckRanges>
+    {
         fn into_event(self) -> builder::Frame {
-            builder::Frame::Ack {}
+            builder::Frame::Ack {
+                ack_delay: Duration::from_micros(self.ack_delay.as_u64()).into_event(),
+                ecn_counts: self.ecn_counts.map(|val| val.into_event()),
+                largest_acknowledged: self.largest_acknowledged().into_event(),
+                ack_range_count: self.ack_ranges().len() as u64,
```

Comment:
can the ack ranges apart from just len be exposed?

Reply:
They are as an event per range. But exposing them in the frame event would require a allocation (either stack or heap) and we really want to avoid that for events.

Reply:
this is the event specifically https://github.com/aws/s2n-quic/blob/main/quic/s2n-quic-events/events/connection.rs#L113-L119